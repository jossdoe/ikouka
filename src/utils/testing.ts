// This is a proxy-file to allow for global wrappers and modifications
// to the functionality of React Testing Library.
// https://testing-library.com/docs/react-testing-library/setup#custom-render
export * from '@testing-library/react'
