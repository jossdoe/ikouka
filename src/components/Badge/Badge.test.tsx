import React from 'react'
import { render, screen } from '../../utils/testing'
import { Badge, BadgeProps } from './index'

const LABEL_TEXT = 'Badge Text'

const COLORS: BadgeProps['color'][] = [
  'gray',
  'green',
  'red',
  'purple',
  undefined
]

const VARIANTS: BadgeProps['variant'][] = ['filled', 'outline', undefined]

const SIZES: BadgeProps['size'][] = ['md', 'lg', undefined]

COLORS.forEach((color) => {
  VARIANTS.forEach((variant) => {
    SIZES.forEach((size) => {
      test(`Badge â€“ ${color}, ${variant}, ${size}`, async () => {
        const { asFragment } = render(
          <Badge color={color} variant={variant} size={size}>
            {LABEL_TEXT}
          </Badge>
        )

        expect(asFragment()).toMatchSnapshot()

        expect(screen.getByText(LABEL_TEXT)).toBeInTheDocument()
      })
    })
  })
})
