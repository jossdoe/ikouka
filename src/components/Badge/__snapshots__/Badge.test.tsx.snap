// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge – gray, filled, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – gray, filled, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – gray, filled, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – gray, outline, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-gray-300 bg-transparent text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – gray, outline, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-gray-300 bg-transparent text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – gray, outline, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-gray-300 bg-transparent text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – gray, undefined, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – gray, undefined, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – gray, undefined, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, filled, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-green-100 text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, filled, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-green-100 text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, filled, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-green-100 text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, outline, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-green-300 bg-transparent text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, outline, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-green-300 bg-transparent text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, outline, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-green-300 bg-transparent text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, undefined, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-green-100 text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, undefined, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-green-100 text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – green, undefined, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-green-100 text-green-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, filled, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-violet-100 text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, filled, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-violet-100 text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, filled, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-violet-100 text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, outline, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-violet-300 bg-transparent text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, outline, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-violet-300 bg-transparent text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, outline, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-violet-300 bg-transparent text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, undefined, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-violet-100 text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, undefined, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-violet-100 text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – purple, undefined, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-violet-100 text-violet-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, filled, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-red-100 text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, filled, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-red-100 text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, filled, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-red-100 text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, outline, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-red-300 bg-transparent text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, outline, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-red-300 bg-transparent text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, outline, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-red-300 bg-transparent text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, undefined, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-red-100 text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, undefined, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-red-100 text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – red, undefined, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-red-100 text-red-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, filled, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, filled, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, filled, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, outline, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-gray-300 bg-transparent text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, outline, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-gray-300 bg-transparent text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, outline, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-gray-300 bg-transparent text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, undefined, lg 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-sm border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, undefined, md 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;

exports[`Badge – undefined, undefined, undefined 1`] = `
<DocumentFragment>
  <span
    class="border uppercase font-semibold py-0.5 px-1.5 rounded text-xs border-transparent bg-gray-100 text-gray-600 "
  >
    Badge Text
  </span>
</DocumentFragment>
`;
